lightning/restore/restore.go

```diff
@@ -461,12 +462,34 @@ func (rc *RestoreController) restoreTables(ctx context.Context) error {
 	stopPeriodicActions := make(chan struct{}, 1)
 	go rc.runPeriodicActions(ctx, stopPeriodicActions)
 
+	type task struct {
+		tr *TableRestore
+		cp *TableCheckpoint
+	}
+	taskCh := make(chan task, rc.cfg.App.IndexConcurrency)
+	defer close(taskCh)
+	for i := 0; i < rc.cfg.App.IndexConcurrency; i++ {
+		go func() {
+			for task := range taskCh {
+				err := task.tr.restoreTable(ctx, rc, task.cp)
+				metric.RecordTableCount("completed", err)
+				restoreErr.Set(task.tr.tableName, err)
+				wg.Done()
+			}
+		}()
+	}
+
 	for _, dbMeta := range rc.dbMetas {
 		dbInfo, ok := rc.dbInfos[dbMeta.Name]
 		if !ok {
 			common.AppLogger.Errorf("database %s not found in rc.dbInfos", dbMeta.Name)
 			continue
 		}
+		// Put the small table in the front of the slice which can avoid large table
+		// take a long time to import and block small table to release index worker.
+		sort.Slice(dbMeta.Tables, func(i, j int) bool {
+			return dbMeta.Tables[i].TotalSize < dbMeta.Tables[j].TotalSize
+		})
```

Comment:
Could you move this inside `(*mdLoaderSetup).setup`? Also update the comment of that function.

https://github.com/pingcap/tidb-lightning/blob/43fdfc3c3cfdfaff2f468662e89a59c422e202f1/lightning/mydump/loader.go#L134-L138